docs/job-scheduling.md

```diff
@@ -8,30 +8,167 @@ featureMaturity:
+
+```bash
+$ cat options.json
+{
+    "service": {
+        "role": "dispatcher"
+    }
+}
+$ dcos package install spark --options=options.json
+```
+
+## Setting Quota for the Executors
+        
+It is recommended to allocate Quota for Spark job executors.  Allocating Quota for the Spark executors provides:
+1.  A guarantee that Spark jobs will receive the requested amount of resources.
+1.  Additional assurance that even misconfigured (e.g. greedy Driver with unset `spark.cores.max`) Spark jobs do not consume
+    too many resources impacting other tenants on the cluster.
+
+The drawback to allocating Quota to the Executors is that Quota resources cannot be used by other frameworks in the
+cluster.
+
+Quota can be allocated for Spark executors in the same way it was allocated for Spark dispatchers.  If we assume we want
+to be able to run 100 executors concurrently each with 1.0 cpu and 4096 MB of memory we should do the following:
+
+```bash
+$ cat executor-quota.json
+{
+  "role": "executor",
+  "guarantee": [
+    {
+      "name": "cpus",
+      "type": "SCALAR",
+      "scalar": { "value": 100.0 }
+    },
+    {
+      "name": "mem",
+      "type": "SCALAR",
+      "scalar": { "value": 409600.0 }
+    }
+  ]
+}
+$ curl -d @executor-quota.json -X POST http://<master>:5050/quota
+
+```
+
+When Spark jobs are submitted they must indicate the role for which the Quota has been set in order to consume resources
+from this quota, for example:
+
+```bash
+$ dcos spark run --verbose --name=spark --submit-args="\
+--driver-cores=1 \
+--driver-memory=1024M \
+--conf spark.cores.max=8 \
+--conf spark.mesos.role=executor \
+--class org.apache.spark.examples.SparkPi \
+https://downloads.mesosphere.com/spark/assets/spark-examples_2.10-1.4.0-SNAPSHOT.jar 3000"
+
+```
+
+**Special consideration for streaming and long-running Spark jobs**: to prevent a single long-running or streaming Spark
+job from consuming the entire Quota the max CPUs for that Spark job should be set to roughly one “job’s worth” of the
+Quota’s resources. This ensures that the Spark job will get sufficient resources to make progress, setting the max CPUs
+ensures it will not starve other Spark jobs of resources as well as predictable offer suppression semantics.
+
+# Setting `spark.cores.max`
+
+To improve Spark job execution reliability, set the maximum number of cores consumed by any given job.  This avoids
+any particular Spark job consuming too many resources in a cluster.  It is highly recommended that each Spark job be
+submitted with a limitation on the maximum number of cores (cpus) it can consume. This is especially important for
+long-running and streaming Spark jobs. 
+
+```bash
+$ dcos spark run --verbose --name=spark --submit-args="\
+--driver-cores=1 \
+--driver-memory=1024M \
+--conf spark.cores.max=8 \ #<< Very important!
+--class org.apache.spark.examples.SparkPi \
+https://downloads.mesosphere.com/spark/assets/spark-examples_2.10-1.4.0-SNAPSHOT.jar 3000"
```

Comment:
Again, the jar version is out of date.

Reply:
Do you suggest that I replace it with a anonymous jar? The nice thing is that now these examples will actually _run_.

Reply:
I like runnable examples. See above.